### PR TITLE
Tezt: Initial setup

### PR DESCRIPTION
**Pull request by francoisthire**
A proof of concept for using the [tezt](https://tezos.gitlab.io/developer/tezt.html?highlight=tezt) framework to implement integration tests for Dedukti.

Several notes:

- We do not want to move all the tests in a single pass, we first validate the framework
- There is an issue with the opam file of `tezt`. Hence we add manually an `opam install` command that should be removed later on

TODO:

- [x] The `opam` file of `tezt` has an issue and should be fixed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,8 +25,40 @@ you have to run the following command:
...
 forget to edit the `.git-blame-ignore-revs` file.
+
+## Writing tests
+
+Tests was originally written with `bash` scripts. We prefer now to use
+[Tezt](https://tezos.gitlab.io/developer/tezt.html?highlight=tezt)
+which is a test framework originally designed for
+[Tezos](https://tezos.gitlab.io).
+
+We invite and recommand developers to write new integration tests
```
Comment (gabrielhdt) [inline CONTRIBUTING.md:41]: recommend

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,8 +25,40 @@ you have to run the following command:
...
+
+- You can list all the tests with `dune exec tests/main.exe -- --list`
+
+- You can filter tests by tags. For example: `dune exec tests/main.exe -- check`.
+   Tags can be negated or combined (see `dune exec tests/main.exe -- --help`)
+
+- You can execute a single test using its name. For example: `dune
+  exec tests/main.exe -- --title "check 'recursive.dk' succeeds"`
+
+- Reset regression tests with `dune exec tests/main.exe -- --reset-regressions`.
```
Comment (gabrielhdt) [inline CONTRIBUTING.md:56]: What does reset mean? Are there regression and other-than-regression tests?

File: Makefile [hunk 3]
```diff
@@ -58,6 +58,10 @@ RESET_REGRESSION?=false
 tests: bin binaries tests/tests.sh
 	$(Q)./tests/tests.sh $(RESET_REGRESSION)
 
+.PHONY: tezt
+tezt: bin binaries
```
Comment (gabrielhdt) [inline Makefile:62]: Do we need two targets? Don't we need only one test target to launch all tests (to limit entry points that are visible by users)?

File: CONTRIBUTING.md (hunk 2 above)
Comment (francoisthire) [reply CONTRIBUTING.md:56]: It means when the regression output changed, you can reset the regression output with this option. I will clarify this.

File: Makefile (hunk 3 above)
Comment (francoisthire) [reply Makefile:62]: For the CI, I think it is best for the moment. When all the tests will be integrated with `Tezt`, the new target won't be needed anymore.

File: CONTRIBUTING.md (hunk 2 above)
Comment (francoisthire) [reply CONTRIBUTING.md:56]: Fix in ed6d39f5

File: CONTRIBUTING.md (hunk 1 above)
Comment (francoisthire) [reply CONTRIBUTING.md:41]: Fixed in ed6d39f5

**Review by gabrielhdt — APPROVED** [review]

**Pull request merged**